js/ui/form/ui.form.layout_manager.utils.js

```diff
@@ -74,8 +75,8 @@ export function convertToRenderFieldItemOptions({
             editorInputId: itemId,
             editorValidationBoundary,
             editorStylingMode,
-            labelMode: isOutsideLabelMode ? 'hidden' : labelMode,
-            labelText: isOutsideLabelMode ? undefined : labelOptions.text,
+            labelMode: item?.editorOptions?.labelMode ?? (labelMode === 'outside' ? 'hidden' : labelMode),
```

Comment:
Let's refactor this code and transfer the logic into _convertToEditorOptions function

Reply:
I am going to help you with it

Reply:
We refactored this code in pair programming